### PR TITLE
support for sparse input features

### PR DESCRIPTION
**Pull request by yoavg**
This patch adds some support for sparse input vectors, speeding up the training for sparse input vectors considerably. Currently, the option to use sparse inputs apply only to FeedForward networks.

### CODE REVIEW

File: theanets/layers.py [hunk 1]
```diff
@@ -13,6 +14,11 @@
 
 FLOAT = theano.config.floatX
 
+def myDot(x, y):
```
Comment (lmjohns3) [inline theanets/layers.py]: Could we just name this _dot?

**Comment by lmjohns3** [comment]
Fantastic! I'm impressed with how little code this requires.

I could merge this now and update later with a name change, or you could do it if you prefer.

**Comment by yoavg** [comment]
I'm not very familiar with the various github mechanisms, anyhow I just pushed a version with the proposed name change, do with it as you please :)

(I am not a theano expert by any means, and was also impressed by how relatively easy it was to get this to work. I hope there isn't some evil bug lurking there, but in my usage it converges to the same solutions as the non-sparse version, and quite a bit faster.)

**Comment by lmjohns3** [comment]
Looks great. If there are bugs in there then we can sort that out later. Thanks for the new feature!

**Pull request merged**